curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/645cc5925f158b5b33e2698f.md

```diff
@@ -191,8 +191,8 @@ const decimalToBinary = (input) => {
 };
 
 const checkUserInput = () => {
-  if (!numberInput.value || isNaN(parseInt(numberInput.value))) {
-    alert("Please provide a decimal number");
+  if (!numberInput.value || isNaN(parseInt(numberInput.value)) || parseInt(numberInput.value) < 0) {
+    alert("Please provide a decimal number greater than or equal to 0");
     return;
```

Comment:
```suggestion
```
Am i the only one who thinks that this return doesn't do anything ? or ami i missing the bigger picture ? @jdwilkin4 ?

Reply:
it stops the function from continuing when the input is invalid